playbooks/cluster-up.yaml

```diff
@@ -33,76 +40,98 @@
       groups: ec2hosts
     loop: "{{ ec2.instances }}"
 
+  - copy:
+      src: "{{ template_ocp_inventory_file }}"
+      dest: "{{ src_ocp_inventory_file }}"
+
+  - replace:
+      path: "{{ src_ocp_inventory_file }}"
+      regexp: '^(openshift_cloudprovider_aws_access_key=)'
+      replace: "\\1{{ lookup('env', 'AWS_ACCESS_KEY_ID') }}"
+
+  - replace:
+      path: "{{ src_ocp_inventory_file }}"
+      regexp: '^(openshift_cloudprovider_aws_secret_key=)'
+      replace: "\\1{{ lookup('env', 'AWS_SECRET_ACCESS_KEY') }}"
+
+  - replace:
+      path: "{{ src_ocp_inventory_file }}"
+      regexp: '^(openshift_clusterid=)'
+      replace: "\\1{{ tag_name }}"
+
+# Pre-deployment
 - hosts: ec2hosts
   name: wait for ssh to come up
   remote_user: "{{ ssh_login }}"
   gather_facts: false
-  
   tasks:
   - wait_for_connection:
       delay: 20
       timeout: 120 
 
-# Everything after this can be modified to our needs. 
+# OCP deployment
 - hosts: ec2hosts
   name: configuration
-  remote_user:  "{{ ssh_login }}"
+  remote_user: "{{ ssh_login }}"
   gather_facts: True
   become: yes
-  vars:
-    known_hosts_path: /root/.ssh/known_hosts
-    key_path: /root/.ssh/ocp_key
   environment:
-    OCP_CLUSTER_ID: "{{ tag_name }}"
+    ANSIBLE_HOST_KEY_CHECKING: false
     AWS_ACCESS_KEY_ID: "{{ lookup('env', 'AWS_ACCESS_KEY_ID') }}" 
     AWS_ACCESS_SECRET_KEY: "{{ lookup('env', 'AWS_SECRET_ACCESS_KEY') }}"
-    ANSIBLE_HOST_KEY_CHECKING: false
  
   tasks:
   - name: check docker service
     service: name=docker state=started
 
-  - name: Set aws access key
+  - name: send ocp inventory template to host 
+    copy:
+      src: "{{ src_ocp_inventory_file }}" 
+      dest: "{{ dest_ocp_inventory_file }}"
+      owner: root
+      group: root
+      mode: 0644
+
+  - name: configure inventory host
+    replace:
+      path: "{{ dest_ocp_inventory_file }}"
+      regexp: 'PRIVATE_DNS_NAME'
+      replace: "{{ hostvars[inventory_hostname].private_dns_name }}"
+  
+  - name: export aws access key
     lineinfile:
       path: /root/.bashrc
       regexp: 'export AWS_ACCESS_KEY_ID=.*'
       line: "export AWS_ACCESS_KEY_ID={{ lookup( 'env', 'AWS_ACCESS_KEY_ID') }}"
 
-  - name: set aws secret key
+  - name: export aws secret key
```

Comment:
I think you are not technically _exporting_ the secret key until the .bashrc file is sourced. So I guess it's exported once each host is ssh'd into...

Reply:
Fair point, I was using the verb to me "sending the keys to the hosts" rather than the actual bash command.